nova_vm/src/ecmascript/builtins/global_object.rs

```diff
@@ -725,15 +728,180 @@ impl GlobalObject {
     ) -> JsResult<Value> {
         todo!()
     }
+
+    /// ### [19.2.5 parseInt ( string, radix )](https://tc39.es/ecma262/#sec-parseint-string-radix)
+    ///
+    /// This function produces an integral Number dictated by interpretation of
+    /// the contents of string according to the specified radix. Leading white
+    /// space in string is ignored. If radix coerces to 0 (such as when it is
+    /// undefined), it is assumed to be 10 except when the number
+    /// representation begins with "0x" or "0X", in which case it is assumed to
+    /// be 16. If radix is 16, the number representation may optionally begin
+    /// with "0x" or "0X".
     fn parse_int(
-        _agent: &mut Agent,
-        _gc: GcScope<'_, '_>,
+        agent: &mut Agent,
+        mut gc: GcScope<'_, '_>,
 
         _this_value: Value,
-        _: ArgumentsList,
+        arguments: ArgumentsList,
     ) -> JsResult<Value> {
-        todo!()
+        let string = arguments.get(0);
+        let radix = arguments.get(1);
+
+        // OPTIMIZATION: If the string is empty, undefined, null or a boolean, return NaN.
+        if string.is_undefined()
+            || string.is_null()
+            || string.is_boolean()
+            || string.is_empty_string()
+        {
+            return Ok(Value::nan());
+        }
+
+        // OPTIMIZATION: If the string is a number and the radix is 10, return the number.
+        if let Value::Integer(radix) = radix {
+            let radix = radix.into_i64();
+            if radix == 10 && string.is_number() {
+                return Ok(string);
```

Comment:
**issue (nonblocking): I forgot that I had tried this, but this fast path is wrong for non-integers. I'm not 100% sure on this but probably this is equal to a `trunc` call. Eg. `parseInt(Math.PI) === 3` IIRC.